pep-0587.rst

```diff
@@ -47,69 +51,79 @@ New structures (4):
 * ``PyConfig``
 * ``PyInitError``
 * ``PyPreConfig``
-* ``PyWideCharList``
+* ``PyWideStringList``
 
-New functions (9):
+New functions (17):
 
 * ``Py_PreInitialize(config)``
 * ``Py_PreInitializeFromArgs(config, argc, argv)``
 * ``Py_PreInitializeFromWideArgs(config, argc, argv)``
+* ``PyWideStringList_Append(list, item)``
+* ``PyConfig_SetString(config_str, str)``
+* ``PyConfig_SetWideString(config_str, str)``
+* ``PyConfig_SetWideStringFromString(config_str, str)``
```

Comment:
I wonder... can we define "PreInitialize" as "enough initialization to allocate str, tuple, list, dict objects" - which doesn't necessarily support arbitrary encodings, to be clear - but then we could use regular C APIs for defining the configuration and simplify this API (possibly even down to an `eval` equivalent if people really want to just do that ;) ).

Again, let's chat :)

Reply:
If possible, I would prefer to postpone this feature for Python 3.9. IMHO the code isn't ready, and this topic wasn't discussed much so far. At least, the PEP 587 design should allow to extend the API to support this use case later. Like: add a new field to PyConfig and new methods to "finish" the "initialization".